test/extended/testdata/router/router-h2spec.yaml

```diff
@@ -0,0 +1,287 @@
+      TY7WWu0gnNI7eq9v+Ck9BtlAAQOvBWK7hPWgooO++PTXgN4SGDAXA2q3capHpGbWoym8WutcZBrP
+      XcW6E7J1Rh6nga3Umt7B4HQ1M1kv+T/oniPGu+Ms/PM/EHVocQMCVsYyo5JuqsNoq3hY4fMZch+T
+      8J2TGkYvrjF9StjiMf7OdFjrOo8QiVWTx68VFGfa7QpOr/hVi9IcDlesbiwypoC2lhP1DEZL8Z4m
+      w4SbZxB2PMilJZDLYJ8UxQZ5R+w1fM4m4/5c6pEaZFWqX2XVmwCDWtAHlAxvTVzbmFeU3QmnhFGC
+      ykvHjLw+2KatoB7v7qVoN/GLsHsMij
+      </body>
+      </html>
+- apiVersion: v1
+  kind: Pod
+  metadata:
+    name: h2spec-haproxy
+    labels:
+      app: h2spec-haproxy
+  spec:
+    containers:
+    - image: ${HAPROXY_IMAGE}
+      name: serve
+      command: ["/bin/bash", "-c" ]
+      args:
+        - set -e;
+          cat /etc/service-certs/tls.key /etc/service-certs/tls.crt > /tmp/bundle.pem;
+          haproxy -f /etc/haproxy/haproxy.config -db
+      ports:
+      - containerPort: 8443
+        protocol: TCP
+      - containerPort: 8080
+        protocol: TCP
+      volumeMounts:
+      - name: service-certs
+        mountPath: /etc/service-certs
+      - name: config
+        mountPath: /etc/haproxy
+      - name: tmp
+        mountPath: /var/cache/haproxy
+      - name: tmp
+        mountPath: /var/run
+    volumes:
+    - name: config
+      configMap:
+        name: h2spec-haproxy-config
+    - name: service-certs
+      secret:
+        secretName: service-certs
+    - name: tmp
+      emptyDir: {}
+    - name: tmp2
+      emptyDir: {}
+- apiVersion: v1
+  kind: Service
+  metadata:
+    name: h2spec-haproxy
+    annotations:
+      service.beta.openshift.io/serving-cert-secret-name: service-certs
+  spec:
+    selector:
+      app: h2spec-haproxy
+    ports:
+      - port: 443
+        name: https
+        targetPort: 8443
+        protocol: TCP
+      - port: 80
+        name: http
+        targetPort: 8080
+        protocol: TCP
+- apiVersion: route.openshift.io/v1
+  kind: Route
+  metadata:
+    labels:
+      app: h2spec-haproxy
+    name: h2spec-haproxy-edge
+  spec:
+    port:
+      targetPort: 8080
+    tls:
+      termination: edge
+    to:
+      kind: Service
+      name: h2spec-haproxy
+      weight: 100
+    wildcardPolicy: None
+- apiVersion: route.openshift.io/v1
+  kind: Route
+  metadata:
+    labels:
+      app: h2spec-haproxy
+    name: h2spec-haproxy-reencrypt
+  spec:
+    port:
+      targetPort: 8443
+    tls:
+      termination: reencrypt
+    to:
+      kind: Service
+      name: h2spec-haproxy
+      weight: 100
+    wildcardPolicy: None
+- apiVersion: route.openshift.io/v1
+  kind: Route
+  metadata:
+    labels:
+      app: h2spec-haproxy
+    name: h2spec-haproxy-passthrough
+  spec:
+    port:
+      targetPort: 8443
+    tls:
+      termination: passthrough
+    to:
+      kind: Service
+      name: h2spec-haproxy
+      weight: 100
+    wildcardPolicy: None
+- apiVersion: v1
+  kind: Pod
+  metadata:
+    name: h2spec
+    labels:
+      app: h2spec
+  spec:
+    containers:
+    - name: h2spec
+      image: docker.io/summerwind/h2spec:2.4.0
```

Comment:
Is coming up with a non-dockerhub image here the last remaining issue?

Reply:
Good point. Does it mean that other images we pull in tests (e.g., nginx) come from another registry?

Reply:
The consensus on Slack was that "in general it’s fine to use an image to get the first 80% of verification".